module/zfs/dmu_traverse.c

```diff
@@ -308,16 +314,41 @@ traverse_visitbp(traverse_data_t *td, const dnode_phys_t *dnp,
 
 		czb = kmem_alloc(sizeof (zbookmark_phys_t), KM_SLEEP);
 
+		/*
+		 * recursively visitbp() blocks below this.
```

Comment:
I'd suggest leaving this one line above the `SET_BOOKMARK` on line 351, and maybe rewording this comment slightly.  How about:

```
		/*
		 * When performing a traversal it is beneficial to
		 * asynchronously read-ahead the upcoming indirect
		 * blocks since they will be needed shortly.  However,
		 * since a 128k indirect (non-L0) block may contain up
		 * to 1024 128-byte block pointers it preferable to not
		 * prefetch them all at once.  Issuing a large number of
		 * async reads may effect performance, and the earlier
		 * the indirect blocks are prefetched the less likely
		 * they are to still be resident in the ARC when needed.
		 * Therefore, prefetching indirect blocks is limited to
		 * zfs_traverse_indirect_prefetch_limit=32 blocks by
		 * default..
		 *
		 * pidx: Index for which next prefetch to be issued.
		 * ptidx: Index at which next prefetch to be triggered.
		/*
```

Reply:
Ok.